src/apps/service/apps.clj

```diff
@@ -210,16 +210,28 @@
     (format-job-submission-response job-info)))
 
 (defn update-job-status
-  ([external-id status end-date]
-     (let [{job-id :job_id} (jobs/get-unique-job-step external-id)]
-       (update-job-status job-id external-id status end-date)))
+  ([external-id]
+   (let [{job-id :job_id} (jobs/get-unique-job-step external-id)
+         updates          (jp/get-job-status-updates external-id)]
+     (when (seq updates)
+       (let [job-step    (jobs/lock-job-step job-id external-id)
+             job         (jobs/lock-job job-id)
+             batch       (when-let [parent-id (:parent_id job)] (jp/get-job-by-id parent-id))
+             apps-client (get-apps-client-for-username (:username job))]
+         (doseq [{:keys [status sent-on]} updates]
+           (let [end-date (when (jp/completed? status) (str sent-on))
+                 job-step (jobs/get-unique-job-step external-id)]
```

Comment:
Not sure if important, but `(jobs/get-unique-job-step external-id)` is also called on line 214.  Not sure if that previous data can be reused.

Reply:
I wanted to get the most recent status of the current job step, and I figured that querying the database again would be much easier and far less error-prone than updating the existing map.